rrmngmnt/firewall.py

```diff
@@ -0,0 +1,93 @@
+from rrmngmnt.host import Host
+from rrmngmnt.user import RootUser
+import shlex
+
+IPTABLES = 'iptables'
+
+
+class Firewall(Host):
+    """
+    Class for firewall services
+    """
+    def __init__(self, ip, root_password):
+        """
+        Args:
+            ip (str): ip or resolvable fqdn
+            root_password (str): root password for host
+        """
+        super(Firewall, self).__init__(ip)
+        self.users.append(RootUser(root_password))
+
+    def check_firewall_service(self, service_name):
+        """
+        Function to check if specific firewall service is running on host
+
+        Args:
+            service_name (str): Name of firewall service to check
+
+        Returns:
+            bool: True if service is running, False otherwise
+
+        """
+        cmd = 'systemctl status {0}'.format(service_name)
+        return not self.run_command(command=shlex.split(cmd))[0]
+
+    def config_firewall(
+        self, service, action, chain, dest, target, protocol='all', ports=None
+    ):
+        """
+        Changes firewall configuration
```

Comment:
I would make interface little bit different ... I have something like this in mind:

h = Host(....)
h.firewall.chain("OUTPUT").add_rule(proto="tcp", traget="xx", port=yy, action="DROP")

and other methods, remove, list ...

I would create class for Chain and Rule, and do all validation and composition of command there.
Then you can add methods to test on active rules.

This is just suggestion, feel free to adjust it according to your needs